File: generators/birdfolkNames.py
__all__ = ['birdfolkNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm11', 'nm12', 'nm4', 'nameGen', 'nm5', 'nm8', 'nm14', 'nm3', 'nm13', 'nm7', 'nm9', 'nm10', 'nm2', 'nm15', 'nm6'])
@Js
def PyJsHoisted_nameGen_(type, this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this, 'type':type}, var)
    var.registers(['tp', 'result', 'type'])
    var.put('tp', var.get('type'))
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            if PyJsStrictEq(var.get('tp'),Js(1.0)):
                var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm6').get('length'))|Js(0.0)))
                var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm8').get('length'))|Js(0.0)))
                var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm10').get('length'))|Js(0.0)))
                if (var.get('i')<Js(3.0)):
                    while (var.get('rnd')<Js(5.0)):
                        var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm6').get('length'))|Js(0.0)))
                    while (var.get('rnd3')<Js(5.0)):
                        var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm10').get('length'))|Js(0.0)))
                    var.put('names', ((var.get('nm6').get(var.get('rnd'))+var.get('nm8').get(var.get('rnd2')))+var.get('nm10').get(var.get('rnd3'))))
                else:
                    var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm7').get('length'))|Js(0.0)))
                    var.put('rnd5', ((var.get('Math').callprop('random')*var.get('nm7').get('length'))|Js(0.0)))
                    var.put('rnd4', ((var.get('Math').callprop('random')*var.get('nm9').get('length'))|Js(0.0)))
                    if (var.get('i')<Js(8.0)):
                        var.put('names', ((((var.get('nm6').get(var.get('rnd'))+var.get('nm7').get(var.get('rnd2')))+var.get('nm9').get(var.get('rnd4')))+var.get('nm7').get(var.get('rnd5')))+var.get('nm10').get(var.get('rnd3'))))
                    else:
                        var.put('rnd7', ((var.get('Math').callprop('random')*var.get('nm7').get('length'))|Js(0.0)))
                        var.put('rnd6', ((var.get('Math').callprop('random')*var.get('nm9').get('length'))|Js(0.0)))
                        while PyJsStrictEq(var.get('nm9').get(var.get('rnd4')),var.get('nm9').get(var.get('rnd6'))):
                            var.put('rnd6', ((var.get('Math').callprop('random')*var.get('nm9').get('length'))|Js(0.0)))
                        var.put('names', ((((((var.get('nm6').get(var.get('rnd'))+var.get('nm7').get(var.get('rnd2')))+var.get('nm9').get(var.get('rnd4')))+var.get('nm7').get(var.get('rnd5')))+var.get('nm9').get(var.get('rnd6')))+var.get('nm7').get(var.get('rnd7')))+var.get('nm10').get(var.get('rnd3'))))
            else:
                if PyJsStrictEq(var.get('tp'),Js(2.0)):
                    var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm11').get('length'))|Js(0.0)))
                    var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm13').get('length'))|Js(0.0)))
                    var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm15').get('length'))|Js(0.0)))
                    if (var.get('i')<Js(3.0)):
                        while (var.get('rnd')<Js(5.0)):
                            var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm11').get('length'))|Js(0.0)))
                        while (var.get('rnd3')<Js(5.0)):
                            var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm15').get('length'))|Js(0.0)))
                        var.put('names', ((var.get('nm11').get(var.get('rnd'))+var.get('nm13').get(var.get('rnd2')))+var.get('nm15').get(var.get('rnd3'))))
                    else:
                        var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm12').get('length'))|Js(0.0)))
                        var.put('rnd5', ((var.get('Math').callprop('random')*var.get('nm12').get('length'))|Js(0.0)))
                        var.put('rnd4', ((var.get('Math').callprop('random')*var.get('nm14').get('length'))|Js(0.0)))
                        if (var.get('i')<Js(8.0)):
                            var.put('names', ((((var.get('nm11').get(var.get('rnd'))+var.get('nm12').get(var.get('rnd2')))+var.get('nm14').get(var.get('rnd4')))+var.get('nm12').get(var.get('rnd5')))+var.get('nm15').get(var.get('rnd3'))))
                        else:
                            var.put('rnd7', ((var.get('Math').callprop('random')*var.get('nm12').get('length'))|Js(0.0)))
                            var.put('rnd6', ((var.get('Math').callprop('random')*var.get('nm14').get('length'))|Js(0.0)))
                            while PyJsStrictEq(var.get('nm14').get(var.get('rnd4')),var.get('nm14').get(var.get('rnd6'))):
                                var.put('rnd6', ((var.get('Math').callprop('random')*var.get('nm14').get('length'))|Js(0.0)))
                            var.put('names', ((((((var.get('nm11').get(var.get('rnd'))+var.get('nm12').get(var.get('rnd2')))+var.get('nm14').get(var.get('rnd4')))+var.get('nm12').get(var.get('rnd5')))+var.get('nm14').get(var.get('rnd6')))+var.get('nm12').get(var.get('rnd7')))+var.get('nm15').get(var.get('rnd3'))))
                else:
                    var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm1').get('length'))|Js(0.0)))
                    var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm3').get('length'))|Js(0.0)))
                    var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm5').get('length'))|Js(0.0)))
                    if (var.get('i')<Js(3.0)):
                        while (var.get('rnd')<Js(5.0)):
                            var.put('rnd', ((var.get('Math').callprop('random')*var.get('nm1').get('length'))|Js(0.0)))
                        while (var.get('rnd3')<Js(5.0)):
                            var.put('rnd3', ((var.get('Math').callprop('random')*var.get('nm5').get('length'))|Js(0.0)))
                        var.put('names', ((var.get('nm1').get(var.get('rnd'))+var.get('nm3').get(var.get('rnd2')))+var.get('nm5').get(var.get('rnd3'))))
                    else:
                        var.put('rnd2', ((var.get('Math').callprop('random')*var.get('nm2').get('length'))|Js(0.0)))
                        var.put('rnd5', ((var.get('Math').callprop('random')*var.get('nm2').get('length'))|Js(0.0)))
                        var.put('rnd4', ((var.get('Math').callprop('random')*var.get('nm4').get('length'))|Js(0.0)))
                        if (var.get('i')<Js(8.0)):
                            var.put('names', ((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm4').get(var.get('rnd4')))+var.get('nm2').get(var.get('rnd5')))+var.get('nm5').get(var.get('rnd3'))))
                        else:
                            var.put('rnd7', ((var.get('Math').callprop('random')*var.get('nm2').get('length'))|Js(0.0)))
                            var.put('rnd6', ((var.get('Math').callprop('random')*var.get('nm4').get('length'))|Js(0.0)))
                            while PyJsStrictEq(var.get('nm4').get(var.get('rnd4')),var.get('nm4').get(var.get('rnd6'))):
                                var.put('rnd6', ((var.get('Math').callprop('random')*var.get('nm4').get('length'))|Js(0.0)))
                            var.put('names', ((((((var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2')))+var.get('nm4').get(var.get('rnd4')))+var.get('nm2').get(var.get('rnd5')))+var.get('nm4').get(var.get('rnd6')))+var.get('nm2').get(var.get('rnd7')))+var.get('nm5').get(var.get('rnd3'))))
            var.get('result').callprop('push', var.get('names'))
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js(''), Js(''), Js(''), Js('c'), Js('cr'), Js('g'), Js('gr'), Js('k'), Js('kr'), Js('q'), Js('qr'), Js('r'), Js('sk'), Js('skr'), Js('sq'), Js('sqr'), Js('sc'), Js('z'), Js('zr')]))
var.put('nm2', Js([Js('a'), Js('i'), Js('e'), Js('o'), Js('u'), Js('a'), Js('i'), Js('i'), Js('a'), Js('e'), Js('e'), Js('a'), Js('i'), Js('e'), Js('o'), Js('u'), Js('a'), Js('i'), Js('i'), Js('a'), Js('e'), Js('e'), Js('ai'), Js('ee'), Js('ia'), Js('ie'), Js('oo'), Js('ua')]))
var.put('nm3', Js([Js('ai'), Js('ee'), Js('ia'), Js('ie'), Js('oo'), Js('ua')]))
var.put('nm4', Js([Js('b'), Js('c'), Js('cc'), Js('k'), Js('n'), Js('nn'), Js('r'), Js('rr'), Js('s'), Js('ss'), Js('shr'), Js('t'), Js('y'), Js('w')]))
var.put('nm5', Js([Js(''), Js(''), Js('k'), Js('q'), Js('t'), Js('th'), Js('w'), Js('ww'), Js('wk')]))
var.put('nm6', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js('c'), Js('cr'), Js('g'), Js('gh'), Js('gr'), Js('k'), Js('kh'), Js('q'), Js('qh'), Js('r'), Js('rh'), Js('sq'), Js('sc'), Js('z'), Js('zr'), Js('zh')]))
var.put('nm7', Js([Js('a'), Js('i'), Js('e'), Js('o'), Js('u'), Js('a'), Js('i'), Js('i'), Js('a'), Js('e'), Js('e'), Js('a'), Js('i'), Js('e'), Js('o'), Js('u'), Js('a'), Js('i'), Js('i'), Js('a'), Js('e'), Js('e'), Js('ee'), Js('ea'), Js('ei'), Js('ia'), Js('ie'), Js('oo')]))
var.put('nm8', Js([Js('ee'), Js('ea'), Js('ei'), Js('ia'), Js('ie'), Js('oo')]))
var.put('nm9', Js([Js('b'), Js('bb'), Js('c'), Js('l'), Js('ll'), Js('n'), Js('nn'), Js('r'), Js('rr'), Js('s'), Js('ss'), Js('sh'), Js('t'), Js('y'), Js('w')]))
var.put('nm10', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js('h'), Js('k'), Js('n'), Js('t'), Js('th'), Js('w'), Js('ww')]))
var.put('nm11', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js('c'), Js('cr'), Js('g'), Js('gh'), Js('gr'), Js('k'), Js('kr'), Js('kh'), Js('q'), Js('qr'), Js('qh'), Js('sk'), Js('r'), Js('rh'), Js('skr'), Js('sq'), Js('sqr'), Js('sc'), Js('z'), Js('zr'), Js('zh')]))
var.put('nm12', Js([Js('a'), Js('i'), Js('e'), Js('o'), Js('u'), Js('a'), Js('i'), Js('i'), Js('a'), Js('e'), Js('e'), Js('a'), Js('i'), Js('e'), Js('o'), Js('u'), Js('a'), Js('i'), Js('i'), Js('a'), Js('e'), Js('e'), Js('ai'), Js('ee'), Js('ea'), Js('ei'), Js('ia'), Js('ie'), Js('oo'), Js('ua')]))
var.put('nm13', Js([Js('ai'), Js('ee'), Js('ea'), Js('ei'), Js('ia'), Js('ie'), Js('oo'), Js('ua')]))
var.put('nm14', Js([Js('b'), Js('bb'), Js('c'), Js('cc'), Js('k'), Js('l'), Js('ll'), Js('n'), Js('nn'), Js('r'), Js('rr'), Js('s'), Js('ss'), Js('sh'), Js('shr'), Js('t'), Js('y'), Js('w')]))
var.put('nm15', Js([Js(''), Js(''), Js(''), Js(''), Js(''), Js('h'), Js('k'), Js('n'), Js('t'), Js('th'), Js('w'), Js('ww'), Js('wk')]))
pass
pass


# Add lib to the module scope
birdfolkNames = var.to_python()

File: js/birdfolkNames.js
var nm1 = ["","","","c","cr","g","gr","k","kr","q","qr","r","sk","skr","sq","sqr","sc","z","zr"];
var nm2 = ["a","i","e","o","u","a","i","i","a","e","e","a","i","e","o","u","a","i","i","a","e","e","ai","ee","ia","ie","oo","ua"];
var nm3 = ["ai","ee","ia","ie","oo","ua"];
var nm4 = ["b","c","cc","k","n","nn","r","rr","s","ss","shr","t","y","w"];
var nm5 = ["","","k","q","t","th","w","ww","wk"];

var nm6 = ["","","","","","c","cr","g","gh","gr","k","kh","q","qh","r","rh","sq","sc","z","zr","zh"];
var nm7 = ["a","i","e","o","u","a","i","i","a","e","e","a","i","e","o","u","a","i","i","a","e","e","ee","ea","ei","ia","ie","oo"];
var nm8 = ["ee","ea","ei","ia","ie","oo"];
var nm9 = ["b","bb","c","l","ll","n","nn","r","rr","s","ss","sh","t","y","w"];
var nm10 = ["","","","","","h","k","n","t","th","w","ww"];

var nm11 = ["","","","","","c","cr","g","gh","gr","k","kr","kh","q","qr","qh","sk","r","rh","skr","sq","sqr","sc","z","zr","zh"];
var nm12 = ["a","i","e","o","u","a","i","i","a","e","e","a","i","e","o","u","a","i","i","a","e","e","ai","ee","ea","ei","ia","ie","oo","ua"];
var nm13 = ["ai","ee","ea","ei","ia","ie","oo","ua"];
var nm14 = ["b","bb","c","cc","k","l","ll","n","nn","r","rr","s","ss","sh","shr","t","y","w"];
var nm15 = ["","","","","","h","k","n","t","th","w","ww","wk"];

var br = "";

function nameGen(type){
	$('#placeholder').css('textTransform', 'capitalize');
	var tp = type;
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		if(tp === 1){
			rnd = Math.random() * nm6.length | 0;
			rnd2 = Math.random() * nm8.length | 0;
			rnd3 = Math.random() * nm10.length | 0;
			if(i < 3){
				while(rnd < 5){
					rnd = Math.random() * nm6.length | 0;
				}
				while(rnd3 < 5){
					rnd3 = Math.random() * nm10.length | 0;
				}
				names = nm6[rnd] + nm8[rnd2] + nm10[rnd3];
			}else{
				rnd2 = Math.random() * nm7.length | 0;
				rnd5 = Math.random() * nm7.length | 0;
				rnd4 = Math.random() * nm9.length | 0;
				if(i < 8){
					names = nm6[rnd] + nm7[rnd2] + nm9[rnd4] + nm7[rnd5] + nm10[rnd3];
				}else{
					rnd7 = Math.random() * nm7.length | 0;
					rnd6 = Math.random() * nm9.length | 0;
					while(nm9[rnd4] === nm9[rnd6]){
						rnd6 = Math.random() * nm9.length | 0;
					}
					names = nm6[rnd] + nm7[rnd2] + nm9[rnd4] + nm7[rnd5] + nm9[rnd6] + nm7[rnd7] + nm10[rnd3];
				}
			}
		}else if(tp === 2){
			rnd = Math.random() * nm11.length | 0;
			rnd2 = Math.random() * nm13.length | 0;
			rnd3 = Math.random() * nm15.length | 0;
			if(i < 3){
				while(rnd < 5){
					rnd = Math.random() * nm11.length | 0;
				}
				while(rnd3 < 5){
					rnd3 = Math.random() * nm15.length | 0;
				}
				names = nm11[rnd] + nm13[rnd2] + nm15[rnd3];
			}else{
				rnd2 = Math.random() * nm12.length | 0;
				rnd5 = Math.random() * nm12.length | 0;
				rnd4 = Math.random() * nm14.length | 0;
				if(i < 8){
					names = nm11[rnd] + nm12[rnd2] + nm14[rnd4] + nm12[rnd5] + nm15[rnd3];
				}else{
					rnd7 = Math.random() * nm12.length | 0;
					rnd6 = Math.random() * nm14.length | 0;
					while(nm14[rnd4] === nm14[rnd6]){
						rnd6 = Math.random() * nm14.length | 0;
					}
					names = nm11[rnd] + nm12[rnd2] + nm14[rnd4] + nm12[rnd5] + nm14[rnd6] + nm12[rnd7] + nm15[rnd3];
				}
			}
		}else{
			rnd = Math.random() * nm1.length | 0;
			rnd2 = Math.random() * nm3.length | 0;
			rnd3 = Math.random() * nm5.length | 0;
			if(i < 3){
				while(rnd < 5){
					rnd = Math.random() * nm1.length | 0;
				}
				while(rnd3 < 5){
					rnd3 = Math.random() * nm5.length | 0;
				}
				names = nm1[rnd] + nm3[rnd2] + nm5[rnd3];
			}else{
				rnd2 = Math.random() * nm2.length | 0;
				rnd5 = Math.random() * nm2.length | 0;
				rnd4 = Math.random() * nm4.length | 0;
				if(i < 8){
					names = nm1[rnd] + nm2[rnd2] + nm4[rnd4] + nm2[rnd5] + nm5[rnd3];
				}else{
					rnd7 = Math.random() * nm2.length | 0;
					rnd6 = Math.random() * nm4.length | 0;
					while(nm4[rnd4] === nm4[rnd6]){
						rnd6 = Math.random() * nm4.length | 0;
					}
					names = nm1[rnd] + nm2[rnd2] + nm4[rnd4] + nm2[rnd5] + nm4[rnd6] + nm2[rnd7] + nm5[rnd3];
				}
			}
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(names));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}